docs/tutorials/customSessionKeys/createSession.md

```diff
@@ -0,0 +1,253 @@
+```
+
+We store the private key in the file using following method. This saves the signer in the file.
+
+```typescript
+const sessionSigner = ethers.Wallet.createRandom();
+const sessionKeyEOA = await sessionSigner.getAddress();
+await sessionFileStorage.addSigner(sessionSigner)
+```
+
+We use the Session Key Manager Module to create a session module using the module address and smart account address and the custom session storage. This is an important relationship to establish - the module provided by the SDK gives you an easy way to interact with modules you write on a smart contract with whatever arbitrary validation logic you need.
+
+
+```typescript
+const sessionModule = await SessionKeyManagerModule.create({
+	moduleAddress: DEFAULT_SESSION_KEY_MANAGER_MODULE,
+	smartAccountAddress: address,
+	sessionStorageClient: sessionFileStorage
+});
+
+```
+
+now, we will cover a deployed contract that validates specific permissions to execute ERC20 token transfers. Using this ERC20 Validation module you will be able to create a dApp that allows user to send a limited amount of funds to a specific address without needing to sign a transaction every single time. In the following example user can only transfer 50 tokens at once.
+
+```typescript
+const sessionKeyData = defaultAbiCoder.encode(
+	["address", "address", "address", "uint256"],
+	[sessionKeyEOA,
+	"0xdA5289fCAAF71d52a80A254da614a192b693e977", // erc20 token address
+	"0x322Af0da66D00be980C7aa006377FCaaEee3BDFD", // receiver address
+	ethers.utils.parseUnits("50".toString(), 6).toHexString(), // 50 usdc amount
+	]
+);
+
+```
+
+next we create the session data. We specify how long this should be valid until or how long it is valid after. This should be a unix timestamp to represent the time. Passing 0 on both makes this session never expire, do not do this in production. Next we pass the module address, session key address, and pass the session key data we just created.
+
+
+```typescript
+const sessionTxData = await sessionModule.createSessionData([{
+	validUntil: 0,
+	validAfter: 0,
+	sessionValidationModule: erc20ModuleAddr,
+	sessionPublicKey: sessionKeyEOA,
+	sessionKeyData: sessionKeyData,
+}]);
+```
+We're going to be tracking if the session key module is already enabled. If we need to enable session key module, we create a transaction using the getEnableModuleData and pass the session key manager module address and push this to the array.
+
+
+```typescript
+const isEnabled = await smartAccount.isModuleEnabled(DEFAULT_SESSION_KEY_MANAGER_MODULE)
+if (!isEnabled) {
+	const enableModuleTrx = await smartAccount.getEnableModuleData(DEFAULT_SESSION_KEY_MANAGER_MODULE);
+	transactionArray.push(enableModuleTrx);
+}
+		
+```
+
+Next we will build a userOp and use the smart account to send it to Bundler.
+
+```typescript
+let partialUserOp = await smartAccount.buildUserOp(transactionArray, {
+	paymasterServiceData: {
```

Comment:
if someone blindly uses this, 
i. They must have passed paymaster instance in SA configuration
ii. The paymaster url used to create should have a gas tank setup on the dashboard
iii. If no policies are added this will work (since it means all allowed). but if some unrelated policy is added then it will throw smart contract address not found. [ remember the one gets currently interpreted as callGasLimit not defined? that case (yes we will fix that)] so they will need to add policy with session key manager address as contract and method setMerkleRoot on it. once registered on the dashboard above code will work for Biconomy SA V2 (when using 3.1.1 SDK)

Reply:
everything is there in the tutorial as well, will add more details here.